content/blog/2021/10/new-architecture-to-ease-kubewarden-administrators-lives.md

```diff
@@ -0,0 +1,257 @@
+
+{{< figure class="center" src="/images/how-it-works-kubewarden.svg" width="90%" alt="new architecture diagram">}}
+
+On the diagram, notice the 2 separate PolicyServer Deployments in cyan and mauve
+(right), created as specified in the 2 PolicyServer resources (left).
+
+Each policy server loads different policies, specified in the
+ClusterAdmissionPolicies. The new PolicyServer Custom Resource is cluster-wide,
+which means that is identifiable by its unique `name`. Here is an example of a
+PolicyServer named `tenant-a`:
+
+```yaml
+---
+apiVersion: policies.kubewarden.io/v1alpha2
+kind: PolicyServer
+metadata:
+  name: tenant-a
+spec:
+  image: ghcr.io/kubewarden/policy-server:latest
+  replicas: 1
+```
+
+A ClusterAdmissionPolicy targeting that PolicyServer needs to set
+`spec.policyServer` to `tenant-a`, as such:
+
+```yaml
+---
+apiVersion: policies.kubewarden.io/v1alpha2
+kind: ClusterAdmissionPolicy
+metadata:
+  name: psp-capabilities
+spec:
+  policyServer: tenant-a
+  module: registry://ghcr.io/kubewarden/policies/psp-capabilities:v0.1.3
+  rules:
+    - apiGroups: [""]
+      apiVersions: ["v1"]
+      resources: ["pods"]
+      operations:
+        - CREATE
+        - UPDATE
+  mutating: true
+  settings:
+    allowed_capabilities:
+      - CHOWN
+    required_drop_capabilities:
+      - NET_ADMIN
+```
+
+For a more in depth dive, have a look at our [architecture
+docs](https://docs.kubewarden.io/architecture.html), and the [Kubewarden CRDs documentation](https://github.com/kubewarden/kubewarden-controller/blob/main/docs/crds/README.asciidoc).
+
+## Ok, but what does this mean for administrators?
+
+Moving from 1 PolicyServer to several increases resiliency and segregation. Now,
+we can take admission reviews from policies and separate them into several
+PolicyServers. While the old architecture was already HA, a noisy
+tenant/namespace or a frequently used policy could bring to a crawl the only
+policy server and break havoc in the Cluster. With the new possibility of
+several PolicyServers, a Kubernetes Operator can isolate policy evaluations per
+tenant/namespace, and run mission critical policies separately, making the whole
+infrastructure more resilient.
+
+The new architecture also validates and mutates PolicyServers and
+ClusterAdmissionPolicies with dedicated admission controllers, for a better UX.
+Which means that administrators can rest comfortably when editing them, as
+catastrophic outcomes (such as all policies being dropped by a misconfigured
+PolicyServer, leading to DOS against the cluster) can never happen. Also,
+ClusterAdmissionPolicies will, if no `spec.policyServer` is defined, bind to the
+PolicyServer named `default`. In addition,
+[Finalizers](https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/)
+are now added to all Kubewarden CRs, which ensure orderly deletion by the
+Kubewarden Controller.
+
+The inclusion of validating and mutating webhooks for Kuberwarden CRDs means that
```

Comment:
I'm split on whether we should use "Custom Resource Definition" instead of "CRD" in the blog post. I think is more readable, but I also understand that whoever is reading this (should probably?) know about the acronym...

Reply:
We can always say Custom Resource Definitions (CRDs) and then use CRD onwards. It's just *so* verbose..

Reply:
Done so in https://github.com/kubewarden/kubewarden.io/pull/33/commits/7068889cfe285fef872928e69ebb3f471dd4e329.